packages/hasher/src/repoInfo.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  WorkspaceInfo,
+  ParsedLock,
+  findGitRoot,
+  getWorkspaces,
+  parseLockFile
+} from "workspace-tools";
+
+import { getPackageDeps } from "@rushstack/package-deps-hash";
+
+export interface RepoInfo {
+  root: string;
+  workspaceInfo: WorkspaceInfo;
+  parsedLock: ParsedLock;
+  repoHashes: { [key: string]: string };
+}
+
+const repoInfoCache: RepoInfo[] = [];
+
+/**
+ * repoInfo cache lookup - it is specialized to be using a substring match to make it run as fast as possible
+ * @param packageRoot
+ */
+function searchRepoInfoCache(packageRoot: string) {
+  for (const repoInfo of repoInfoCache) {
+    if (repoInfo.workspaceInfo && packageRoot.startsWith(repoInfo.root)) {
+      return repoInfo;
+    }
+  }
+}
+
+export async function _getRepoInfoNoCache(cwd: string) {
+  const root = findGitRoot(cwd);
+  if (!root) {
+    throw new Error("Cannot initialize Repo class without a .git root");
+  }
+
+  const repoHashes = getPackageDeps(root).files;
```

Comment:
Just a thought: this tool uses "git ls-tree" by spawning a new process, implementing 'git ls-tree' in JS should be trivial and remove one process spawning + output parsing (`git status` and `git hash-object` are probably best left to the git implementaion).

Reply:
I think this will need to be vetted to see if it's indeed faster. I have had tremendous luck with a repo with 20k files that can do this + do hashing in less than 1s. `git ls-tree` is a very fast process vs having to use a recursive glob of all the files inside node.js. What is good here is that this implementation detail is hidden from backfill, so it can reap the benefit of that research on a later date if it IS faster to do node.js fs globbing.

Reply:
I not thinking about globbing but doing the same as git does but in JavaScript (look up recursively the git object tree). But this was more of a minor random thought.